src/smt/theory_str_noodler/quant_lia_solver.h

```diff
@@ -42,11 +45,12 @@ namespace smt::noodler {
 
             // another options for a solver: mk_smt_solver(m, p, symbol("LIA")); (no tactic)
             // tactic solver used by z3 to solve quantified LIA formula
-            solver* sl = mk_tactic2solver(m, mk_lia_tactic(m, p), p, false, true, false, symbol("ALL"));
+            solver* sl = mk_tactic2solver(m, mk_lia_tactic(m, p), p, false, true, true, symbol("ALL"));
 
             erv.push_back(e);
             sl->assert_expr(erv);
             auto res = sl->check_sat();
+            sl->get_unsat_core(unsat_core);
```

Comment:
Is it possible this will make it slower?

Reply:
Why is it necessary?

Maybe we could at least move it to `get_unsat_core()`, add some bool member that will remember whether we need to `pop_back()` and if another `assert()` or `check_sat()` is called, we first pop back.